phpBB/phpbb/datetime.php

```diff
@@ -117,13 +117,23 @@ public function format($format = '', $force_absolute = false)
 					if ($day !== false)
 					{
 						// Format using the short formatting and finally swap out the relative token placeholder with the correct value
-						return str_replace(self::RELATIVE_WRAPPER . self::RELATIVE_WRAPPER, $this->user->lang['datetime'][$day], strtr(parent::format($format['format_short']), $format['lang']));
+						return str_replace(self::RELATIVE_WRAPPER . self::RELATIVE_WRAPPER, $this->user->lang['datetime'][$day], strtr(substr(preg_replace_callback("/(\P{Latin})([AP]M|[ap]m)(\P{Latin})/u",
+								function ($matches)
+								{
+									global $user;
```

Comment:
Why `global $user;` ? Isn't the User object already available in this class? Moreover, it would be better to use the Language object nowadays for this kind of stuffs.